src/index.js

```diff
@@ -38,16 +37,18 @@ function addPropsTable(storyFn, context, infoOptions) {
       exclude: options.propTablesExclude,
       order: options.propTablesSortOrder,
       children: storyFn
-    }),
+    }).map(c => ({ ...c })),
```

Comment:
Perhaps this looks odd, but what it's doing is serializing the function object into something usable once it goes through the event emitter.

Typically, component function objects will have a `__docgenInfo` property consumed by this addon.

When you pass functions through storybook's channel event emitter, you still get the function, but it's properties are undefined-- which is useless for what we're trying to do here.

Iterating the enumerable properties of the function object --by spreading the function object into a new object-- ensures `__docgenInfo` will be defined. 😅